module/VuFindSearch/src/VuFindSearch/Backend/Blender/Backend.php

```diff
@@ -0,0 +1,618 @@
+                        break;
+                    }
+                }
+            }
+
+            if (null === $record) {
+                break;
+            }
+            $mergedCollection->add($record, false);
+        }
+    }
+
+    /**
+     * Retrieve a single document.
+     *
+     * @param string   $id     Document identifier
+     * @param ParamBag $params Search backend parameters
+     *
+     * @return \VuFindSearch\Response\RecordCollectionInterface
+     */
+    public function retrieve($id, ParamBag $params = null)
+    {
+        throw new \Exception('Blender does not support retrieve');
+    }
+
+    /**
+     * Return the record collection factory.
+     *
+     * Lazy loads a generic collection factory.
+     *
+     * @return RecordCollectionFactoryInterface
+     */
+    public function getRecordCollectionFactory()
+    {
+        throw new \Exception('getRecordCollectionFactory not supported in Blender');
+    }
+
+    /**
+     * Get active backends for a search
+     *
+     * @param ParamBag $params Search backend parameters
+     *
+     * @return array
+     */
+    protected function getActiveBackends(ParamBag $params): array
+    {
+        $activeBackends = $this->backends;
+
+        // Handle the blender_backend pseudo-filter
+        $fq = $params->get('fq') ?? [];
+        $filteredBackends = [];
+        foreach ($fq as $filter) {
+            $advanced = preg_match(
+                '/\{!tag=blender_backend_filter}blender_backend:\((.+)\)/',
+                $filter,
+                $matches
+            );
+            if ($advanced) {
+                $filter = explode(' OR ', $matches[1]);
+            }
+            foreach ((array)$filter as $current) {
+                if (strncmp($current, 'blender_backend:', 16) === 0) {
+                    $active = trim(substr($current, 16), '"');
+                    if (!isset($activeBackends[$active])) {
+                        $this->logWarning(
+                            "Invalid blender_backend filter: Backend $active not"
+                            . ' enabled'
+                        );
+                    } else {
+                        $filteredBackends[$active] = $activeBackends[$active];
+                    }
+                }
+            }
+        }
+        if ($filteredBackends) {
+            $activeBackends = $filteredBackends;
+        }
+        foreach ($fq as $current) {
+            if (strncmp($current, '-blender_backend:', 17) === 0) {
+                $disabled = trim(substr($current, 17), '"');
+                if (isset($activeBackends[$disabled])) {
+                    unset($activeBackends[$disabled]);
+                }
+            }
+        }
+
+        return $activeBackends;
+    }
+
+    /**
+     * Get next record from the given backend.
+     *
+     * Gets next records from the previously retrieved array of records or retrieves
+     * a new batch of records from.
```

Comment:
I don't think this sentence is complete -- from what?

Reply:
Fixed!